osutil/testhelper.go

```diff
@@ -38,3 +38,11 @@ func MustBeTestBinary(panicMsg string) {
 		panic(panicMsg)
 	}
 }
+
+// A checker for the logic that shall not run in a test context
+// ! NOTE: Use this judiciously as it might impose a test-skip risk
+func CheckScopeIsNotTestExecution(panicMsg string) {
```

Comment:
nitpick: since we have `MustBeTestBinary()`, I'd call this `MustNotBeTestBinary()`

Reply:
Thank you.
Will certainly do that.